### PR TITLE
SC-3704 - Analog-consent

### PR DESCRIPTION
**Pull request by phnm**
# Description

UI implementation of the bulk analog consent flow. Implements a table action for the student administration table. The main addition is the new administration/students/consent page. The new bulk-consent store takes care of passing along the selected student IDs of the index page to the new consent page.

## Links to Tickets or other pull requests
https://ticketsystem.schul-cloud.org/browse/SC-3706

## Changes

- TableDataRow now passes the rowindex prop along
- added the generatePassword mixin from legacy client
- birthday and password fields are changeble
- fixed some bugs
- fixed some $store objects
- fixed consent icons in the student data table
- filtering users who have already consented
- add validation to birthday input
- creating dynamically a pdf file for downloading and printing

## Screenshots of UI changes

image
image
Screenshot 2020-08-12 at 14 25 07
Screenshot 2020-08-12 at 14 25 34
Screenshot 2020-08-12 at 14 26 33

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: src/pages/administration/students/consent.vue [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+			</backend-data-table>
+			<div id="consent-checkbox">
+				<base-input v-model="check" type="checkbox" name="switch" label="">
+				</base-input>
+				<label>
+					<i18n
+						path="pages.administration.students.consent.steps.register.confirm"
+					>
+						<template v-slot:analogConsent>
+							<a class="link" :href="fileLinks.analogConsent" target="_">{{
```
Comment (qnowak) [inline src/pages/administration/students/consent.vue:99]: Is this target needed? Or maybe this should be "_blank" ?

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Comment by murat-merdoglu-hpi** [comment]
@schul-cloud-bot update screenshots

**Review by j-jmnz — APPROVED** [review]

**Pull request merged**